File: backend/models/database/character.go
package database

import (
	"encoding/json"
	"time"
)

type Character struct {
	ID         string          `json:"id"`
	GameID     string          `json:"game_id"`
	UserID     *string         `json:"user_id"` // Pointer because it can be NULL
	Name       string          `json:"name"`
	Race       string          `json:"race"`
	MaxHP      int             `json:"max_hp"`
	CurrentHP  int             `json:"current_hp"`
	AvatarURL  *string         `json:"avatar_url"`
	Stats      json.RawMessage `json:"stats"`
	Inventory  json.RawMessage `json:"inventory"`
	IsNPC      bool            `json:"is_npc"`
	Money      int             `json:"money"`
	CreatedAt  time.Time       `json:"created_at"`
	PlayerName *string         `json:"player_name,omitempty"` // Populated when joining with user table

	// New fields
	Initiative int             `json:"initiative"`
	Age        string          `json:"age"`
	Height     string          `json:"height"`
	Weight     string          `json:"weight"`
	MaxSpells  int             `json:"max_spells"`
	Spells     json.RawMessage `json:"spells"`
	Abilities  string          `json:"abilities"`
	Experience int             `json:"experience"`
}

File: src/lib/types/character.ts
export interface Character {
    id: string;
    game_id: string;
    user_id?: string | null;
    name: string;
    race: string;
    max_hp: number;
    current_hp: number;
    avatar_url?: string | null;
    stats: Record<string, { value: number; modifier: number }>;
    inventory: InventoryItem[];
    is_npc: boolean;
    money: number;
    created_at: string;
    player_name?: string;
    type: string;
    sub_race?: string | null;

    // New fields
    initiative: number;
    age?: string | null;
    height?: string | null;
    weight?: string | null;
    max_spells?: number;
    spells?: Record<string, { name: string; description: string; charges: string }[]>;
    abilities?: string | null;
    experience?: number;
    armor_class: number;
    speed: number;
    conditions: string[];
}

export interface InventoryItem {
    name: string;
    quantity: string;
    description?: string;
    image_url?: string;
    icon_name?: string;
}
